### PR TITLE
Rewrite of LisOption

### PR DESCRIPTION
**Pull request by chleh**
I largely rewrote the class `LisOption`. Now any Lis option can be given in the input file. The new implementation `LisOption` just stores options. `LisLinearSolver` then passes them through to Lis.

I simplified setting of options: There are no longer any conditions that decide what will happen. I also removed unused methods, members and enums.

### CODE REVIEW

File: MathLib/LinAlg/Lis/LisLinearSolver.cpp [hunk 1]
```diff
@@ -70,62 +42,64 @@ void LisLinearSolver::solve(LisVector &b, LisVector &x)
 
     INFO("------------------------------------------------------------------");
     INFO("*** LIS solver computation");
-#ifdef _OPENMP
```
Comment (wenqing) [inline MathLib/LinAlg/Lis/LisLinearSolver.cpp]: Looks great. Only one question hereby: where to print number of threads used by LIS if this ifdef is removed?

Comment (chleh) [inline MathLib/LinAlg/Lis/LisLinearSolver.cpp]: I see: This can be an interesting fact to know. I will re-enable it.

Comment (chleh) [inline MathLib/LinAlg/Lis/LisLinearSolver.cpp]: Re-Introduced OpenMP output. See below.

Comment (wenqing) [inline MathLib/LinAlg/Lis/LisLinearSolver.cpp]: :+1:

**Comment by wenqing** [comment]
:fast_forward:

**Comment by endJunction** [comment]
Merged.

**Pull request closed without merging**